"Issue a token" tutorial

mDuo13 opened the pull request:
Adds a new tutorial that shows how to issue a fungible token on the XRP Ledger Testnet, including interactive examples and working sample code. Also:

- Fixes a style issue with interactive tutorials
- Fixes some bugs in the submit-and-verify code (the version already in the repo wrongly concludes a transaction has permanently failed sometimes when it could still be validated later). Removes an old, alternative example of submit and verify from the RippleAPI quickstart and links¹ the new (fixed) version instead.
- Adds a new tutorial category, "Use Tokens" to the Tutorials. This is the only tutorial in that category so far, but I hope to make more soon. (For example, tutorials on buying or selling a token in the exchange, sending a token to another user via cross-currency payment or rippling, issuing NFTs using one standard or another, etc. Details on those are TBD at a later date.)

¹A couple of the links to sample code will be broken until after this is merged. Until then, the sample code is at: https://github.com/XRPLF/xrpl-dev-portal/tree/issue_token_tutorial/content/_code-samples/issue-a-token and https://github.com/XRPLF/xrpl-dev-portal/tree/issue_token_tutorial/content/_code-samples/submit-and-verify

mDuo13 committed:
Fix interactive tutorial breadcrumb spacing

mDuo13 committed:
Start Issue a Token tutorial

mDuo13 committed:
Issue Token: settings recs

mDuo13 committed:
Issue Token: start mocking up UI

mDuo13 committed:
Fungible token: form validation UI, refinements

mDuo13 committed:
Issue a token: start code sample

mDuo13 committed:
Issue a token: working code

- Fix bugs in submit and verify helper function
- Tweak tutorial and placement of helper function
- Finish drafting sample code
- Put sample code into tutorial

mDuo13 committed:
Issue a token: interactive & node.js compatible

- Make sample code node.js compatible (need to uncomment a few lines)
- Wire up interactive tutorial bits to actually run the relevant code
- Edit instructions slightly, add notes on Require Destination Tags

mDuo13 commented:
(Rebased on latest master)

mDuo13 committed:
Issue token tutorial: use memos, wait steps"

mDuo13 committed:
Fix spelling and link errors

mDuo13 committed:
Issue a token: fix some Chrome stylings

mDuo13 committed:
Issue a Token improvements: python, UI fixes, etc.

- Display some errors without marking the step as complete
  (Also applied to Require Destination Tags tutorial)
- Add Python equivalent script

mDuo13 committed:
Issue a Token: fix link in JA target

mDuo13 committed:
Issue a token: add README to sample code

mvadari commented on content/_code-samples/issue-a-token/js/demo.html:
```diff
@@ -0,0 +1,12 @@
+<meta charset="utf-8" />
+<title>Code Sample - Issue a Token</title>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.21/lodash.min.js" integrity="sha512-WFN04846sdKMIP5LKNphMaWzU7YpMyCU245etK3g/2ARYbPK9Ub18eG+ljU96qKRCWh+quCY7yefSmlkQw1ANQ==" crossorigin="anonymous"></script>
+<script src="https://unpkg.com/ripple-lib@1.9.8/build/ripple-latest-min.js"></script>
```
This can be updated to 1.10

mvadari commented on content/_code-samples/issue-a-token/js/issue-a-token.js:
```diff
@@ -0,0 +1,194 @@
+  // This doesn't technically need to happen after you call api.connect() but
+  // it's convenient to do here.
+  async function get_faucet_address() {
+    const faucet_url = "https://faucet.altnet.rippletest.net/accounts"
```
There's a built-in method to do this now in ripple-lib: https://github.com/ripple/ripple-lib/pull/1497

mDuo13 replied:
Updated the samples to use the new function.

mDuo13 committed:
Update ripple-lib samples to 1.10.0

- Removes lodash dependecy for web version
- Uses the new generateFaucetWallet() API method
- Make 'issue a token' code capable of running in-browser or in Node.js
  without modification

mDuo13 committed:
Fix require destination tags code sample

mvadari commented on content/_code-samples/issue-a-token/js/issue-a-token.js:
```diff
@@ -0,0 +1,177 @@
+
+  // Get credentials from the Testnet Faucet -----------------------------------
+  console.log("Requesting addresses from the Testnet faucet...")
+  const hot_data = await api.generateFaucetWallet()
+  const hot_address = hot_data.account.classicAddress
+  const hot_secret = hot_data.account.secret
+
+  const cold_data = await api.generateFaucetWallet()
+  const cold_address = cold_data.account.classicAddress
+  const cold_secret = cold_data.account.secret
```
Would it be better to use terms like `issuer_address` and `receiver_address`?

mDuo13 replied:
I spent a lot of time thinking about this at the outset and decided that it's better to use the cold/hot nomenclature since that's widespread in the industry, it's good practice, and the terms are also nice and short.

nkramer44 committed:
add issue token tutorial code for xrpl4j

mDuo13 committed:
Merge pull request #1154 from nkramer44/nk/xrpl4j-issue-token

Add issue token example code for xrpl4j

mDuo13 committed:
Issue a token: Incorporate Java code, clarify some details

mDuo13 committed:
Issue a token: minor cleanup
